tools/tck-api/src/test/scala/org/opencypher/tools/tck/values/CypherValueTest.scala

```diff
@@ -107,4 +109,47 @@ class CypherValueTest extends AnyFunSuite with Matchers {
     l2 should equal(l1)
   }
 
+  test("node comparison with labelled nodes") {
+    CypherNode(Set("A", "B")) should equal(CypherNode(Set("B", "A")))
+    CypherNode(Set("A", "C")) should not equal(CypherNode(Set("A", "B")))
+  }
+
+  test("list comparison with labelled nodes") {
+    val nodeList1 = List(
+      CypherNode(scala.collection.immutable.SortedSet("A", "D")),
+      CypherNode(scala.collection.immutable.SortedSet("B", "C"))
+    )
+    val nodeList2 = List(
+      CypherNode(scala.collection.immutable.SortedSet("D", "A")(Ordering.String.reverse)),
+      CypherNode(scala.collection.immutable.SortedSet("C", "B")(Ordering.String.reverse))
+    )
+
+    CypherOrderedList(nodeList1) should equal(CypherOrderedList(nodeList2))
+    CypherOrderedList(nodeList2) should equal(CypherOrderedList(nodeList1))
+    CypherUnorderedList(nodeList1) should equal(CypherUnorderedList(nodeList2))
+    CypherUnorderedList(nodeList2) should equal(CypherUnorderedList(nodeList1))
+    CypherOrderedList(nodeList1) should equal(CypherUnorderedList(nodeList2))
+    CypherOrderedList(nodeList2) should equal(CypherUnorderedList(nodeList1))
+    CypherUnorderedList(nodeList1) should equal(CypherOrderedList(nodeList2))
+    CypherUnorderedList(nodeList2) should equal(CypherOrderedList(nodeList1))
+
+    CypherUnorderedList(nodeList1.reverse) should equal(CypherUnorderedList(nodeList2))
```

Comment:
Would an unordered list1 in reverse order be equal to order list2 or not. We be good to add these test cases.

Reply:
The way it's designed it should be equal, added some more test cases. Thanks!